### PR TITLE
[Life Domains] Rework Domain Content Rendering

### PR DESCRIPTION
**Pull request by MarkDavies0**
## Changes
adds `react-native-render-html` for displaying domain importance html strings as RN views. See Firestore/domains 'Sleep' domain for an example

## Requirements
Run `yarn install` in /mobile

## PR Dependencies

Depends on:  #136 

## Preview
See notion `Content Formatting`

## Testing

- [x] Manual

### CODE REVIEW

**Comment by wynnset** [comment]
As discussed, please fix the left and right margins on the domain screen to be the same from top to bottom.

**Comment by wynnset** [comment]
As discussed, let's enter the HTML data properly and just do a text find/replace to create the structure necessary to display properly (replace `<ul></ul>` with `p` tags instead and change `<li>` to bullet point + space and change `</li>` to `br`?)

**Comment by wynnset** [comment]
@MarkDavies0 hey sorry just assigning this back since you didn't answer my question.

**Comment by wynnset** [comment]
One more small UI thing: can we please reduce the side margins beside the embedded panels as well so they match the margins for the text too? Both the "what to know" panel and the strategies panels at the bottom have increased margins. (Sorry app crashed locally when I tried to take screenshot of strategies but the margins are there if you look)
image

**Comment by wynnset** [comment]
More issues from local testing:
Console warning:
image
Subdomain content looks off:
image
Tapping on the strategy inside of a subdomain does nothing:
image
If 4 strategies are already selected, can we hide the unchecked checkboxes to indicate they have reached the max:
image
Got stuck here:
image

File: mobile/src/helpers/DomainHelper.tsx [hunk 1]
```diff
@@ -71,29 +70,11 @@ export function sum(elements: number[]): number {
...
-    }
-    return str;
-  })
-}
\ No newline at end of file
+export function replaceListTags(htmlString: string): string {
+  let newString = htmlString.replace(new RegExp('<ul>', 'g'), '<p>');
+  newString = newString.replace(new RegExp('<li>', 'g'), '• ');
+  newString = newString.replace(new RegExp('</li>', 'g'), '<br>');
+  newString = newString.replace(new RegExp('<br></ul>', 'g'), '</p>'); // remove the <br> after the last bullet point
```
[the remark this answers is not in the document]
Comment (wynnset) [reply mobile/src/helpers/DomainHelper.tsx:77]: As discussed, let's go ahead and use the [altering the dom tree](https://meliorence.github.io/react-native-render-html/docs/guides/dom-tampering#altering-the-dom-tree) functionality to do this whole thing instead.

**Comment by wynnset** [comment]
Hey @MarkDavies0 I am still seeing some of the issues from before that remain (the ones that are not marked as completed in the previous comment check list). Also:
- [ ] I clicked on the link in the description of the domain and it caused the app to crash
- [ ] The margins are fixed on the domain, but not the subdomain page

**Comment by wynnset** [comment]
Couple more issues found during testing:
image
image
image
image

**Comment by MarkDavies0** [comment]
> * [ ]  I clicked on the link in the description of the domain and it caused the app to crash

@wynnset Do you have a screen shot of the link the caused the crash?

**Comment by wynnset** [comment]
Now I'm hitting a new problem (this happened with 2 different accounts, both with their user localSettings and userState reset as per your instruction):image

**Comment by MarkDavies0** [comment]
@wynnset Fixed

**Comment by wynnset** [comment]
- [ ] The orb labels are using slugs instead of domain names: image
- [ ] Clicking the back button when on Notifications screen after choosing domains does nothing. We should not show the back button if it doesn't do anything: image

**Pull request merged**